coreneuron/io/reports/report_configuration_parser.cpp

```diff
@@ -96,10 +113,45 @@ std::vector<ReportConfiguration> create_report_configurations(const char* conf_f
             if (strcmp(report_on, "i_membrane") == 0) {
                 nrn_use_fast_imem = true;
                 report.type = IMembraneReport;
-            } else if (is_soma)
-                report.type = SomaReport;
-            else
-                report.type = CompartmentReport;
+            } else {
+                switch (target_type) {
+                    case TargetType::Soma:
+                        report.type = SomaReport;
+                        break;
+                    case TargetType::Axon:
+                        report.type = SectionReport;
+                        report.section_type = Axon;
+                        report.section_all_compartments = false;
+                        break;
+                    case TargetType::AxonComp:
+                        report.type = SectionReport;
+                        report.section_type = Axon;
+                        report.section_all_compartments = true;
+                        break;
+                    case TargetType::Dendrite:
+                        report.type = SectionReport;
+                        report.section_type = Dendrite;
+                        report.section_all_compartments = false;
+                        break;
+                    case TargetType::DendriteComp:
+                        report.type = SectionReport;
+                        report.section_type = Dendrite;
+                        report.section_all_compartments = true;
+                        break;
+                    case TargetType::Apical:
+                        report.type = SectionReport;
+                        report.section_type = Apical;
+                        report.section_all_compartments = false;
+                        break;
+                    case TargetType::ApicalComp:
+                        report.type = SectionReport;
+                        report.section_type = Apical;
+                        report.section_all_compartments = true;
+                        break;
+                    default:
+                        report.type = CompartmentReport;
```

Comment:
Minor  - Is there possibility to avoid `default` here? (like Alex suggested. The reason is to avoid surprise like what Sirio had when he specified Section target type)

Reply:
Agreed. Technically, we already filter most of the errors from the Neurodamus-side, but just in case, I completely agree.